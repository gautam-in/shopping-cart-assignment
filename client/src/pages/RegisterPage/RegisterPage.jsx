import React, { useEffect, useState } from "react";
import classes from "./RegisterPage.module.scss";
import { useForm } from "react-hook-form";
import { useHistory } from "react-router-dom";
import { userCred } from "../../Redux/cred";
import { useDispatch, useSelector } from "react-redux";
export default function RegisterPage() {
  const history = useHistory();
  const dispatch = useDispatch();
  let presentEmail = useSelector((state) => state.userList.userCredentials);
  const [loading, setLoading] = useState(false);
  const {
    register,
    formState: { errors },
    handleSubmit,
    getValues,
  } = useForm();

  const onSubmission = (data) => {
    setLoading(true);
    const storeCred = presentEmail.find(
      (user) => user.email === data.email.trim()
    );

    if (storeCred) {
      alert("You are already registered with this email kindly login ");
      history.push("/signin");
    } else {
      dispatch(
        userCred({
          id: new Date().getMilliseconds(),
          email: data.email.trim(),
          password: data.password.trim(),
        })
      );
      alert("Congrats ! You are Registered");
      setLoading(false);

      history.push("/signin");
    }
  };

  useEffect(() => {
    document.title = "S B | Register";
  }, []);

  return (
    <div className={classes.Container}>
      <section className={classes.PrimaryContainer}>
        <h1>Signup</h1>
        <p>We do not share your personal details with anyone.</p>
      </section>
      <section className={classes.SecondaryContainer}>
        <form
          className={classes.Form}
          autoComplete="off"
          onSubmit={handleSubmit(onSubmission)}
        >
          <fieldset
            style={{ border: "none" }}
            disabled={loading}
            aria-busy={loading}
          >
            <div className={classes.InputContainer}>
              <input
                className={classes.Input}
                id="fname"
                {...register("fname")}
                type="text"
                required
                autoComplete="off"
                placeholder="First Name"
              />
              <label className={classes.Label} htmlFor="fname">
                First Name
              </label>

              <p>{errors.fname?.message}</p>
            </div>
            <div className={classes.InputContainer}>
              <input
                id="lname"
                className={classes.Input}
                {...register("lname")}
                type="text"
                autoComplete="off"
                required
                placeholder="Last Name"
              />
              <label className={classes.Label} htmlFor="lname">
                Last Name
              </label>
            </div>
            <div className={classes.InputContainer}>
              <input
                id="email"
                className={classes.Input}
                type="email"
                {...register("email")}
                required
                autoComplete="off"
                placeholder="Email"
              />

              <label className={classes.Label} htmlFor="email">
                Email
              </label>
            </div>
            <div className={classes.InputContainer}>
              <input
                id="password"
                className={classes.Input}
                autoComplete="off"
                type="password"
                {...register("password")}
                required
                placeholder="Password"
              />
              <label className={classes.Label} htmlFor="password">
                Password
              </label>
            </div>
            <div className={classes.InputContainer}>
              <input
                id="passwordConfirmation"
                className={classes.Input}
                autoComplete="off"
                type="password"
                {...register("passwordConfirmation", {
                  required: "Please confirm password!",
                  validate: {
                    matchesPreviousPassword: (value) => {
                      const { password } = getValues();
                      return password === value || "Passwords should match!";
                    },
                  },
                })}
                required
                placeholder="Confirm Password"
              />
              <label className={classes.Label} htmlFor="Repassword">
                Confirm Password
              </label>
              {errors.passwordConfirmation && (
                <p style={{ color: "red" }}>
                  {errors.passwordConfirmation.message}
                </p>
              )}
            </div>
            <div className={classes.InputContainer}>
              <button type="submit" className={classes.Button}>
                Signup
              </button>
            </div>
          </fieldset>
        </form>
      </section>
    </div>
  );
}
