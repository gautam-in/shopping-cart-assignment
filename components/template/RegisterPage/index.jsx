import { Component } from "react";
import { Row, Col, Button } from 'reactstrap';
import PageLayout from "../../organism/PageLayout";
import RegisterForm from "../../organism/RegisterForm";
class RegisterPage extends Component {
    render(){
       return (
          <PageLayout>
             <Row>
                <Col md={6} className="pt-3 pl-5">
                   <h2>Signup</h2>
                   <p>We do not share your personal details with anyone.</p>
                </Col>
                <Col md={6}>
                   <RegisterForm />
                </Col>
             </Row>
          </PageLayout>
       )
    }
}

export default RegisterPage;