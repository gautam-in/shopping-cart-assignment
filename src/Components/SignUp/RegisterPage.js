import React from 'react'
import "./Signup.css"

export default function RegisterPage(){

    return(
        <main>
        
        <section className="signup-section">
            <div className="signup-header">
            <h2>Signup</h2>
            <p>We do not share your personal details with anyone.</p>
            </div>
           
            <form className="singup-form">
           
            <input type="text" placeholder="First Name" required />
            <input type="text" placeholder="Last Name" required />
            <input type="email" placeholder="Email" required />
            <input type="password" placeholder="Password" required />
            <input type="password" placeholder="Confirm Password" required />
            <input type="submit" className="btn" value="Signup"></input>
            
     
        </form>

        </section>
    
    </main>
    )
}