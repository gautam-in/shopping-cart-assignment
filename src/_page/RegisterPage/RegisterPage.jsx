import React from 'react';
import { connect } from 'react-redux';
import { userActions } from '../../_actions';
import Footer from '../../_components/footer/footer';
import Header from '../../_components/header/header';
import '../LoginPage/style.scss';


class RegisterPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                firstName: '',
                lastName: '',
                email: '',
                password: '',
                confirm_password:''
            },
            errors: {},
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        const { name, value } = event.target;
        const { user } = this.state;
        this.setState({
            user: {
                ...user,
                [name]: value
            }
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        this.setState({ submitted: true });
        const { user } = this.state;
        if (this.validate()) {
            this.props.register(user);
        }
    }

    validate(){
        let user = this.state.user;
        let errors = {};
        let isValid = true;

        if (!user["firstName"]) {
            isValid = false;
            errors["firstName"] = "Please enter First Name.";
         }

         if (!user["lastName"]) {
            isValid = false;
            errors["lastName"] = "Please enter Last Name.";
         }
         
        if (typeof user["email"] !== "undefined") {
          var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
          if (!pattern.test(user["email"])) {
            isValid = false;
            errors["email"] = "Please enter valid email address.";
          }
        }

        if (!user["password"]) {
            isValid = false;
            errors["password"] = "Please enter password.";
         }

        if (typeof user["password"] !== "undefined") {
            
            var pattern = new RegExp(/^(?=.*[a-z])(?=.*[0-9])(?=.{7,})/i);
            if (!pattern.test(user["password"])) {
              isValid = false;
              errors["password"] = "Please enter valid password.";
            }
          }
          if (!user["confirm_password"]) {
            isValid = false;
            errors["confirm_password"] = "Please enter confirm password.";
          }
          if (typeof user["password"] !== "undefined" && typeof user["confirm_password"] !== "undefined") {
            if (user["password"] != user["confirm_password"]) {
              isValid = false;
              errors["confirm_password"] = "Passwords don't match.";
            }
        }
        this.setState({
          errors: errors
        });
    
        return isValid;
    }
    render() {
        const { user } = this.state;
        return (
        <div>
           <Header />
            <div className="main">

            <div className="title">

                <h2>SignUp</h2>
                <h5>We don't share personal data with anyone.</h5>
            </div>
            <div className="regForm">

            <form onSubmit={this.handleSubmit}>
            <div >
                <input type="text" name="firstName" placeholder=" " value={user.firstName} onChange={this.handleChange} />
                <label htmlFor="firstName">First Name</label>
                <p >{this.state.errors.firstName}</p>
            </div>
            <div>
                <input type="text" className="form-control" placeholder=" " name="lastName" value={user.lastName} onChange={this.handleChange} />
                <label htmlFor="lastName">Last Name</label>
                <p >{this.state.errors.lastName}</p>
            </div>
            <div>
                <input type="text" className="form-control" placeholder=" " name="email" value={user.email} onChange={this.handleChange} />
                <label htmlFor="email">Email</label>
                <p >{this.state.errors.email}</p>
            </div>
            <div>
                <input type="password" className="form-control" placeholder=" " name="password" value={user.password} onChange={this.handleChange} />
                <label htmlFor="password">Password</label>
                <p >{this.state.errors.password}</p>
                
            </div>
            <div>
                <input type="password" name="confirm_password" placeholder=" " value={user.confirm_password} onChange={this.handleChange} className="form-control" id="confirm_password" />
                <label htmlFor="confirm_password">Confirm Password</label>
                <p >{this.state.errors.confirm_password}</p>
                
            </div>
            <div className="submit">
                <input type="submit" name="" value="SignUp"/>
            </div>
            </form>

            </div>

        </div>
        <div className='regFooter'>
            <Footer />
        </div>
    </div>
        );
    }
}

function mapState(state) {
    const { registering } = state.registration;
    return { registering };
}

const actionCreators = {
    register: userActions.register
}

const connectedRegisterPage = connect(mapState, actionCreators)(RegisterPage);
export { connectedRegisterPage as RegisterPage };