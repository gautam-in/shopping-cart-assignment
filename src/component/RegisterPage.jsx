import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";
import { selectUser } from "../store/slices/user/user.selector";
import Button from "./Button";
import FormInput from "./FormInput";
import "../styles/register.scss";
import { signUpwithEmail } from "../store/slices/user/user.action";

const RegisterPage = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const user = useSelector(selectUser);

  const defaultFormFields = {
    firstName: "",
    lastName: "",
    email: "",
    password: "",
    confirmPassword: "",
  };

  const [formFields, setFormFields] = useState(defaultFormFields);
  const { firstName, lastName, email, password, confirmPassword } = formFields;

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (password !== confirmPassword) {
      alert("password do not match");
      return;
    }
    try {
      dispatch(signUpwithEmail(email, password));
      // const user = await createAuthUserWithEmailAndPassword(email, password);
      user && navigate("/");
      setFormFields(defaultFormFields);
    } catch (error) {
      if (error.code === "auth/email-already-in-use") alert("user exists");
      console.log("user creation failed", error.message);
    }
  };

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormFields({
      ...formFields,
      [name]: value,
    });
  };

  return (
    <form onSubmit={handleSubmit} className="sign-up-container">
      <div className="sign-up-header">
        <h2 className="title">SignUp</h2>
        <span>We do not share your details with anyone.</span>
      </div>
      <div>
        <FormInput
          type="text"
          name="firstName"
          value={firstName}
          label="First Name"
          required
          onChange={handleChange}
        />
        <FormInput
          type="text"
          name="lastName"
          value={lastName}
          label="Last Name"
          required
          onChange={handleChange}
        />
        <FormInput
          type="email"
          name="email"
          value={email}
          label="Email"
          required
          onChange={handleChange}
        />
        <FormInput
          type="password"
          name="password"
          value={password}
          label="Password"
          required
          onChange={handleChange}
          inputMode="numeric"
          pattern="^(?=.*[a-zA-Z])(?=.*[0-9])(?!.* ).{6,}$"
        />
        <FormInput
          type="password"
          name="confirmPassword"
          value={confirmPassword}
          label="Confirm Password"
          required
          onChange={handleChange}
        />
        <div className="buttons-container">
          <Button type="submit" buttonType="max" onSubmit={handleSubmit}>
            Sign In
          </Button>
        </div>
      </div>
    </form>
  );
};

export default RegisterPage;
