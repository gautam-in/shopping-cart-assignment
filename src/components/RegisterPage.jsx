import React, { Component } from 'react';
import RegisterForm from "./RegisterForm";

class RegisterPage extends Component {
  render() {
    return (
        <>
        <div className="container top-container">
          <div className="row">
              <div className="col-md-5 pd-3">
                <h1>Signup</h1>
                <br/>
                <p>
                    We do not share your personal details with anyone.
                </p>
              </div>
              <div className="col-md-5 pd-3">
                <RegisterForm />
              </div>
          </div>
        </div>
        </>
    );
  }
}

export default RegisterPage;