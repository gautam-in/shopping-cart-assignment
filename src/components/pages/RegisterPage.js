import React from 'react';
import formInputs from '../../shared/Register.json';
import * as Constants from '../../shared/constants';
import FormPage from '../templates/FormPage/FormPage';
import { useHistory } from 'react-router-dom';
const RegisterPage = () => {
  const history = useHistory();
  const registerUser = () => {
    history.push('/home');
  };
  return (
    <main className='container'>
      <FormPage
        title={Constants.RegisterTitle}
        desc={Constants.RegisterDescription}
        formInputs={formInputs}
        formSubmit={registerUser}
      />
    </main>
  );
};

export default RegisterPage;
