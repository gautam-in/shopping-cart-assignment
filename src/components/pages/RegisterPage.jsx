import React, { useState, useEffect } from 'react'
import RegisterForm from "../Molecules/RegisterForm";

const RegisterPage = () => {
    useEffect(() => {
      document.title = 'Signup - Sabka Bazar'
  }, []);
    return (
        <>
        <div className="container top-container">
          <div className="row">
              <div className="col-md-5 pd-3">
                <h1>Signup</h1>
                <br/>
                <p>
                    We do not share your personal details with anyone.
                </p>
              </div>
              <div className="col-md-5 pd-3">
                <RegisterForm />
              </div>
          </div>
        </div>
        </>
    );
}

export default RegisterPage;