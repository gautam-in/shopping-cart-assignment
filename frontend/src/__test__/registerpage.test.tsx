import { render, screen } from "@testing-library/react";

import RegisterPage from "@/pages/register";

jest.mock("next/router", () => require("next-router-mock"));

describe("Register", () => {
  it("renders a register page", () => {
    render(<RegisterPage />);

    const heading = screen.getByRole("heading", {
      name: /signup/i,
    });
    expect(heading).toBeInTheDocument();
  });
  it("should match RegisterPage snapshot", () => {
    const wrapper = render(<RegisterPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
