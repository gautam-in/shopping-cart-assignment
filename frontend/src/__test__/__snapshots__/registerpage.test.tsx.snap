// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should match RegisterPage snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="login-section"
      >
        <header>
          <h2>
            Signup
          </h2>
          <p>
            We do not share your personal data with anyone.
          </p>
        </header>
        <form>
          <div
            class="form-input-filed"
          >
            <input
              aria-label="First Name"
              id="firstName"
              placeholder=" "
              type="text"
              value=""
            />
            <label
              class="form-input-filed--text-label"
              for="firstName"
            >
              First Name
            </label>
          </div>
          <div
            class="form-input-filed"
          >
            <input
              aria-label="Last Name"
              id="lastName"
              placeholder=" "
              type="text"
              value=""
            />
            <label
              class="form-input-filed--text-label"
              for="lastName"
            >
              Last Name
            </label>
          </div>
          <div
            class="form-input-filed"
          >
            <input
              aria-label="Email"
              id="email"
              placeholder=" "
              type="email"
              value=""
            />
            <label
              class="form-input-filed--text-label"
              for="email"
            >
              Email
            </label>
          </div>
          <div
            class="form-input-filed"
          >
            <input
              aria-label="Password"
              id="password"
              placeholder=" "
              type="password"
              value=""
            />
            <label
              class="form-input-filed--text-label"
              for="password"
            >
              Password
            </label>
          </div>
          <div
            class="form-input-filed"
          >
            <input
              aria-label="Confirm Password"
              id="confirmPassword"
              placeholder=" "
              type="password"
              value=""
            />
            <label
              class="form-input-filed--text-label"
              for="confirmPassword"
            >
              Confirm Password
            </label>
          </div>
          <button
            class="btn btn--primary undefined"
            disabled=""
            type="submit"
          >
            Signup
          </button>
        </form>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="login-section"
    >
      <header>
        <h2>
          Signup
        </h2>
        <p>
          We do not share your personal data with anyone.
        </p>
      </header>
      <form>
        <div
          class="form-input-filed"
        >
          <input
            aria-label="First Name"
            id="firstName"
            placeholder=" "
            type="text"
            value=""
          />
          <label
            class="form-input-filed--text-label"
            for="firstName"
          >
            First Name
          </label>
        </div>
        <div
          class="form-input-filed"
        >
          <input
            aria-label="Last Name"
            id="lastName"
            placeholder=" "
            type="text"
            value=""
          />
          <label
            class="form-input-filed--text-label"
            for="lastName"
          >
            Last Name
          </label>
        </div>
        <div
          class="form-input-filed"
        >
          <input
            aria-label="Email"
            id="email"
            placeholder=" "
            type="email"
            value=""
          />
          <label
            class="form-input-filed--text-label"
            for="email"
          >
            Email
          </label>
        </div>
        <div
          class="form-input-filed"
        >
          <input
            aria-label="Password"
            id="password"
            placeholder=" "
            type="password"
            value=""
          />
          <label
            class="form-input-filed--text-label"
            for="password"
          >
            Password
          </label>
        </div>
        <div
          class="form-input-filed"
        >
          <input
            aria-label="Confirm Password"
            id="confirmPassword"
            placeholder=" "
            type="password"
            value=""
          />
          <label
            class="form-input-filed--text-label"
            for="confirmPassword"
          >
            Confirm Password
          </label>
        </div>
        <button
          class="btn btn--primary undefined"
          disabled=""
          type="submit"
        >
          Signup
        </button>
      </form>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
